diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/arch/mips/Makefile linuxmips-2.4.30/arch/mips/Makefile
--- linuxmips-2.4.30.ref/arch/mips/Makefile	2007-05-11 16:37:45.000000000 -0700
+++ linuxmips-2.4.30/arch/mips/Makefile	2007-05-15 17:26:55.000000000 -0700
@@ -46,7 +46,7 @@
 GCCFLAGS	:= -I $(TOPDIR)/include/asm/gcc
 GCCFLAGS	+= -G 0 -mno-abicalls -fno-pic -pipe -fno-builtin
 GCCFLAGS	+= $(call check_gcc, -finline-limit=100000,)
-LINKFLAGS	+= -G 0 -static -n
+LINKFLAGS	+= -G 0 -static -n -nostdlib
 MODFLAGS	+= -mlong-calls
 
 ifdef CONFIG_DEBUG_INFO
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/asm-mips/uaccess.h linuxmips-2.4.30/include/asm-mips/uaccess.h
--- linuxmips-2.4.30.ref/include/asm-mips/uaccess.h	2007-05-11 16:37:46.000000000 -0700
+++ linuxmips-2.4.30/include/asm-mips/uaccess.h	2007-05-16 10:57:50.000000000 -0700
@@ -234,51 +234,45 @@
  * for 32 bit mode and old iron.
  */
 #ifdef __mips64
-#define __GET_USER_DW(__gu_err) __get_user_asm("ld", __gu_err)
+#define __GET_USER_DW(val, ptr) __get_user_asm(val, "ld", ptr)
 #else
-#define __GET_USER_DW(__gu_err) __get_user_asm_ll32(__gu_err)
+#define __GET_USER_DW(val, ptr) __get_user_asm_ll32(val, ptr)
 #endif
 
-#define __get_user_nocheck(x,ptr,size)					\
-({									\
-	long __gu_err = 0;						\
-	__typeof(*(ptr)) __gu_val = 0;					\
-	long __gu_addr;							\
-	__gu_addr = (long) (ptr);					\
+#define __get_user_common(val, size, ptr)				\
+do {									\
 	switch (size) {							\
-	case 1: __get_user_asm("lb", __gu_err); break;			\
-	case 2: __get_user_asm("lh", __gu_err); break;			\
-	case 4: __get_user_asm("lw", __gu_err); break;			\
-	case 8: __GET_USER_DW(__gu_err); break;				\
+	case 1: __get_user_asm(val, "lb", ptr); break;			\
+	case 2: __get_user_asm(val, "lh", ptr); break;			\
+	case 4: __get_user_asm(val, "lw", ptr); break;			\
+	case 8: __GET_USER_DW(val, ptr); break;				\
 	default: __get_user_unknown(); break;				\
 	}								\
-	 x = (__typeof__(*(ptr))) __gu_val;				\
-	__gu_err;							\
+} while (0)
+
+#define __get_user_nocheck(x,ptr,size)					\
+({									\
+        long __gu_err;                                                  \
+                                                                        \
+        __get_user_common((x), size, ptr);                              \
+        __gu_err;                                                       \
 })
 
 #define __get_user_check(x,ptr,size)					\
 ({									\
-	__typeof__(*(ptr)) __gu_val = 0;				\
-	long __gu_addr = (long) (ptr);					\
-	long __gu_err;							\
-									\
-	__gu_err = verify_area(VERIFY_READ, (void *) __gu_addr, size);	\
-									\
-	if (likely(!__gu_err)) {					\
-		switch (size) {						\
-		case 1: __get_user_asm("lb", __gu_err); break;		\
-		case 2: __get_user_asm("lh", __gu_err); break;		\
-		case 4: __get_user_asm("lw", __gu_err); break;		\
-		case 8: __GET_USER_DW(__gu_err); break;			\
-		default: __get_user_unknown(); break;			\
-		}							\
-	}								\
-	x = (__typeof__(*(ptr))) __gu_val;				\
-	 __gu_err;							\
+        long __gu_err = -EFAULT;                                        \
+        const __typeof__(*(ptr)) __user * __gu_ptr = (ptr);             \
+                                                                        \
+        if (likely(access_ok(VERIFY_READ,  __gu_ptr, size)))            \
+                __get_user_common((x), size, __gu_ptr);                 \
+                                                                        \
+        __gu_err;                                                       \
 })
 
-#define __get_user_asm(insn,__gu_err)					\
-({									\
+#define __get_user_asm(val, insn, addr)					\
+{									\
+	long __gu_tmp;							\
+									\
 	__asm__ __volatile__(						\
 	"1:	" insn "	%1, %3				\n"	\
 	"2:							\n"	\
@@ -289,21 +283,27 @@
 	"	.section __ex_table,\"a\"			\n"	\
 	"	"__UA_ADDR "\t1b, 3b				\n"	\
 	"	.previous					\n"	\
-	: "=r" (__gu_err), "=r" (__gu_val)				\
-	: "0" (__gu_err), "o" (__m(__gu_addr)), "i" (-EFAULT));		\
-})
+	: "=r" (__gu_err), "=r" (__gu_tmp)				\
+	: "0" (0), "o" (__m(addr)), "i" (-EFAULT));			\
+									\
+	(val) = (__typeof__(*(addr))) __gu_tmp;				\
+}
 
 /*
  * Get a long long 64 using 32 bit registers.
  */
-#define __get_user_asm_ll32(__gu_err)					\
-({									\
+#define __get_user_asm_ll32(val, addr)					\
+{									\
+	union {								\
+		unsigned long long	l;				\
+		__typeof__(*(addr))	t;				\
+	} __gu_tmp;							\
+									\
 	__asm__ __volatile__(						\
-	"1:	lw	%1, %3					\n"	\
-	"2:	lw	%D1, %4					\n"	\
-	"	move	%0, $0					\n"	\
+	"1:	lw	%1, (%3)				\n"	\
+	"2:	lw	%D1, 4(%3)				\n"	\
 	"3:	.section	.fixup,\"ax\"			\n"	\
-	"4:	li	%0, %5					\n"	\
+	"4:	li	%0, %4					\n"	\
 	"	move	%1, $0					\n"	\
 	"	move	%D1, $0					\n"	\
 	"	j	3b					\n"	\
@@ -312,10 +312,11 @@
 	"	" __UA_ADDR "	1b, 4b				\n"	\
 	"	" __UA_ADDR "	2b, 4b				\n"	\
 	"	.previous					\n"	\
-	: "=r" (__gu_err), "=&r" (__gu_val)				\
-	: "0" (__gu_err), "o" (__m(__gu_addr)),				\
-	  "o" (__m(__gu_addr + 4)), "i" (-EFAULT));			\
-})
+	: "=r" (__gu_err), "=&r" (__gu_tmp.l)				\
+	: "0" (0), "r" (addr), "i" (-EFAULT));				\
+									\
+	(val) = __gu_tmp.t;						\
+}
 
 extern void __get_user_unknown(void);
 
@@ -324,23 +325,22 @@
  * for 32 bit mode and old iron.
  */
 #ifdef __mips64
-#define __PUT_USER_DW(__pu_val) __put_user_asm("sd", __pu_val)
+#define __PUT_USER_DW(ptr) __put_user_asm("sd", ptr)
 #else
-#define __PUT_USER_DW(__pu_val) __put_user_asm_ll32(__pu_val)
+#define __PUT_USER_DW(ptr) __put_user_asm_ll32(ptr)
 #endif
 
 #define __put_user_nocheck(x,ptr,size)					\
 ({									\
-	long __pu_err = 0;						\
 	__typeof__(*(ptr)) __pu_val;					\
-	long __pu_addr;							\
+	long __pu_err = 0;						\
+									\
 	__pu_val = (x);							\
-	__pu_addr = (long) (ptr);					\
 	switch (size) {							\
-	case 1: __put_user_asm("sb", __pu_val); break;			\
-	case 2: __put_user_asm("sh", __pu_val); break;			\
-	case 4: __put_user_asm("sw", __pu_val); break;			\
-	case 8: __PUT_USER_DW(__pu_val); break;				\
+	case 1: __put_user_asm("sb", ptr); break;			\
+	case 2: __put_user_asm("sh", ptr); break;			\
+	case 4: __put_user_asm("sw", ptr); break;			\
+	case 8: __PUT_USER_DW(ptr); break;				\
 	default: __put_user_unknown(); break;				\
 	}								\
 	__pu_err;							\
@@ -348,26 +348,24 @@
 
 #define __put_user_check(x,ptr,size)					\
 ({									\
+	__typeof__(*(ptr)) __user *__pu_addr = (ptr);			\
 	__typeof__(*(ptr)) __pu_val = (x);				\
-	long __pu_addr = (long) (ptr);					\
-	long __pu_err;							\
-									\
-	__pu_err = verify_area(VERIFY_WRITE, (void *) __pu_addr, size);	\
+	long __pu_err = -EFAULT;					\
 									\
-	if (likely(!__pu_err)) {					\
+	if (likely(access_ok(VERIFY_WRITE,  __pu_addr, size))) {	\
 		switch (size) {						\
-		case 1: __put_user_asm("sb", __pu_val); break;		\
-		case 2: __put_user_asm("sh", __pu_val); break;		\
-		case 4: __put_user_asm("sw", __pu_val); break;		\
-		case 8: __PUT_USER_DW(__pu_val); break;			\
+		case 1: __put_user_asm("sb", __pu_addr); break;		\
+		case 2: __put_user_asm("sh", __pu_addr); break;		\
+		case 4: __put_user_asm("sw", __pu_addr); break;		\
+		case 8: __PUT_USER_DW(__pu_addr); break;		\
 		default: __put_user_unknown(); break;			\
 		}							\
 	}								\
 	__pu_err;							\
 })
 
-#define __put_user_asm(insn, __pu_val)					\
-({									\
+#define __put_user_asm(insn, ptr)					\
+{									\
 	__asm__ __volatile__(						\
 	"1:	" insn "	%z2, %3		# __put_user_asm\n"	\
 	"2:							\n"	\
@@ -379,18 +377,18 @@
 	"	" __UA_ADDR "	1b, 3b				\n"	\
 	"	.previous					\n"	\
 	: "=r" (__pu_err)						\
-	: "0" (__pu_err), "Jr" (__pu_val), "o" (__m(__pu_addr)),	\
+	: "0" (0), "Jr" (__pu_val), "o" (__m(ptr)),			\
 	  "i" (-EFAULT));						\
-})
+}
 
-#define __put_user_asm_ll32(__pu_val)					\
-({									\
+#define __put_user_asm_ll32(ptr)					\
+{									\
 	__asm__ __volatile__(						\
-	"1:	sw	%2, %3		# __put_user_asm_ll32	\n"	\
-	"2:	sw	%D2, %4					\n"	\
+	"1:	sw	%2, (%3)	# __put_user_asm_ll32	\n"	\
+	"2:	sw	%D2, 4(%3)				\n"	\
 	"3:							\n"	\
 	"	.section	.fixup,\"ax\"			\n"	\
-	"4:	li	%0, %5					\n"	\
+	"4:	li	%0, %4					\n"	\
 	"	j	3b					\n"	\
 	"	.previous					\n"	\
 	"	.section	__ex_table,\"a\"		\n"	\
@@ -398,9 +396,9 @@
 	"	" __UA_ADDR "	2b, 4b				\n"	\
 	"	.previous"						\
 	: "=r" (__pu_err)						\
-	: "0" (__pu_err), "r" (__pu_val), "o" (__m(__pu_addr)),		\
-	  "o" (__m(__pu_addr + 4)), "i" (-EFAULT));			\
-})
+	: "0" (0), "r" (__pu_val), "r" (ptr),				\
+	  "i" (-EFAULT));						\
+}
 
 extern void __put_user_unknown(void);
 
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/linux/fs.h linuxmips-2.4.30/include/linux/fs.h
--- linuxmips-2.4.30.ref/include/linux/fs.h	2007-05-11 16:37:45.000000000 -0700
+++ linuxmips-2.4.30/include/linux/fs.h	2007-05-16 11:17:08.000000000 -0700
@@ -514,7 +514,9 @@
 		struct reiserfs_inode_info	reiserfs_i;
 		struct bfs_inode_info		bfs_i;
 		struct udf_inode_info		udf_i;
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
 		struct ncp_inode_info		ncpfs_i;
+#endif /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
 		struct proc_inode_info		proc_i;
 		struct socket			socket_i;
 		struct usbdev_inode_info        usbdev_i;
@@ -789,7 +791,9 @@
 		struct reiserfs_sb_info	reiserfs_sb;
 		struct bfs_sb_info	bfs_sb;
 		struct udf_sb_info	udf_sb;
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
 		struct ncp_sb_info	ncpfs_sb;
+#endif /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
 		struct usbdev_sb_info   usbdevfs_sb;
 		struct jffs2_sb_info	jffs2_sb;
 		struct cramfs_sb_info	cramfs_sb;
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/linux/ncp_fs.h linuxmips-2.4.30/include/linux/ncp_fs.h
--- linuxmips-2.4.30.ref/include/linux/ncp_fs.h	2001-03-09 12:34:57.000000000 -0800
+++ linuxmips-2.4.30/include/linux/ncp_fs.h	2007-05-16 11:11:43.000000000 -0700
@@ -8,6 +8,8 @@
 #ifndef _LINUX_NCP_FS_H
 #define _LINUX_NCP_FS_H
 
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
+
 #include <linux/fs.h>
 #include <linux/in.h>
 #include <linux/types.h>
@@ -291,4 +293,7 @@
 
 #endif				/* __KERNEL__ */
 
+#endif                          /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
+
 #endif				/* _LINUX_NCP_FS_H */
+
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/linux/ncp_fs_i.h linuxmips-2.4.30/include/linux/ncp_fs_i.h
--- linuxmips-2.4.30.ref/include/linux/ncp_fs_i.h	2000-07-03 14:46:09.000000000 -0700
+++ linuxmips-2.4.30/include/linux/ncp_fs_i.h	2007-05-16 11:14:40.000000000 -0700
@@ -8,6 +8,8 @@
 #ifndef _LINUX_NCP_FS_I
 #define _LINUX_NCP_FS_I
 
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
+
 #ifdef __KERNEL__
 
 /*
@@ -29,4 +31,6 @@
 
 #endif	/* __KERNEL__ */
 
+#endif                          /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
+
 #endif	/* _LINUX_NCP_FS_I */
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/linux/ncp_fs_sb.h linuxmips-2.4.30/include/linux/ncp_fs_sb.h
--- linuxmips-2.4.30.ref/include/linux/ncp_fs_sb.h	2000-07-07 17:53:08.000000000 -0700
+++ linuxmips-2.4.30/include/linux/ncp_fs_sb.h	2007-05-16 11:15:07.000000000 -0700
@@ -8,6 +8,8 @@
 #ifndef _NCP_FS_SB
 #define _NCP_FS_SB
 
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
+
 #include <linux/types.h>
 #include <linux/ncp_mount.h>
 
@@ -101,5 +103,7 @@
 
 #endif				/* __KERNEL__ */
 
+#endif                          /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
+
 #endif
  
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/linux/ncp.h linuxmips-2.4.30/include/linux/ncp.h
--- linuxmips-2.4.30.ref/include/linux/ncp.h	2003-08-13 10:19:29.000000000 -0700
+++ linuxmips-2.4.30/include/linux/ncp.h	2007-05-16 11:13:35.000000000 -0700
@@ -10,6 +10,8 @@
 #ifndef _LINUX_NCP_H
 #define _LINUX_NCP_H
 
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
+
 #include <linux/types.h>
 
 #define NCP_PTYPE                (0x11)
@@ -176,4 +178,6 @@
 	__u32 sequence __attribute__((packed));
 };
 
+#endif                          /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
+
 #endif				/* _LINUX_NCP_H */
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/include/linux/ncp_mount.h linuxmips-2.4.30/include/linux/ncp_mount.h
--- linuxmips-2.4.30.ref/include/linux/ncp_mount.h	2000-02-15 17:07:58.000000000 -0800
+++ linuxmips-2.4.30/include/linux/ncp_mount.h	2007-05-16 11:15:38.000000000 -0700
@@ -8,6 +8,8 @@
 #ifndef _LINUX_NCP_MOUNT_H
 #define _LINUX_NCP_MOUNT_H
 
+#if defined(CONFIG_NCP_FS_MODULE) || defined(CONFIG_NCP_FS)
+
 #include <linux/types.h>
 #include <linux/ncp.h>
 
@@ -86,4 +88,6 @@
 
 #endif /* __KERNEL__ */
 
+#endif                          /* CONFIG_NCP_FS_MODULE || CONFIG_NCP_FS */
+
 #endif
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/Makefile linuxmips-2.4.30/Makefile
--- linuxmips-2.4.30.ref/Makefile	2007-05-11 16:37:46.000000000 -0700
+++ linuxmips-2.4.30/Makefile	2007-05-15 17:30:54.000000000 -0700
@@ -116,7 +116,7 @@
 CPPFLAGS := -D__KERNEL__ -I$(HPATH)
 
 CFLAGS := $(CPPFLAGS) -Wall -Wstrict-prototypes -Wno-trigraphs -O2 \
-	  -fno-strict-aliasing -fno-common
+	  -fno-strict-aliasing -fno-common -ffreestanding
 ifndef CONFIG_FRAME_POINTER
 CFLAGS += -fomit-frame-pointer
 endif
diff -Naur --exclude=CVS --exclude='*.o' --exclude='*.a' --exclude='*.so' --exclude='*.elf' --exclude=System.map --exclude=Makefile.d --exclude='*log' --exclude='*log2' --exclude='*~' --exclude='.*~' --exclude='.#*' --exclude='*.bak' --exclude='*.orig' --exclude='*.rej' --exclude='core.[0-9]*' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=.depend --exclude='.*.o.flags' --exclude='*.gz' --exclude=vmlinux --exclude=vmlinux.bin --exclude=yamon-02.06-SIGMADESIGNS-01_el.bin linuxmips-2.4.30.ref/README.1035.gcc41x.patch linuxmips-2.4.30/README.1035.gcc41x.patch
--- linuxmips-2.4.30.ref/README.1035.gcc41x.patch	1969-12-31 16:00:00.000000000 -0800
+++ linuxmips-2.4.30/README.1035.gcc41x.patch	2007-05-16 11:23:31.000000000 -0700
@@ -0,0 +1,19 @@
+Feature:
+--------
+To support GCC 4.1.x or above.
+
+Prerequisite patch numbers:
+---------------------------
+1016
+1032
+
+Primary author:
+---------------
+YH Lin
+
+Related to which chip version SMP86xx xx=?
+-----------------------------------------
+all
+
+(linux patches) which CONFIG_... are provided:
+----------------------------------------------
