--- binutils/bfd/elf32-mips.c~
+++ binutils/bfd/elf32-mips.c
@@ -1613,7 +1613,9 @@
 
 /* The SVR4 MIPS ABI says that this should be 0x10000, and Linux uses
    page sizes of up to that limit, so we need to respect it.  */
-#define ELF_MAXPAGESIZE			0x10000
+/*#define ELF_MAXPAGESIZE			0x10000*/
+/* Use 4K to shrink the elf header.  NOT for general use! */
+#define ELF_MAXPAGESIZE			0x1000
 #define elf32_bed			elf32_tradbed
 
 /* Include the target file again for this target.  */
--- binutils/bfd/elfn32-mips.c~
+++ binutils/bfd/elfn32-mips.c
@@ -2399,7 +2399,9 @@
 
 /* The SVR4 MIPS ABI says that this should be 0x10000, and Linux uses
    page sizes of up to that limit, so we need to respect it.  */
-#define ELF_MAXPAGESIZE			0x10000
+/*#define ELF_MAXPAGESIZE			0x10000*/
+/* Use 4K to shrink the elf header.  NOT for general use! */
+#define ELF_MAXPAGESIZE			0x1000
 #define elf32_bed			elf32_tradbed
 
 /* Include the target file again for this target.  */
--- binutils/ld/ld.texinfo
+++ binutils/ld/ld.texinfo
@@ -1653,7 +1653,7 @@

 @kindex --version-script=@var{version-scriptfile}
 @cindex version script, symbol versions
-@itemx --version-script=@var{version-scriptfile}
+itemx --version-script= var version-scriptfile
 Specify the name of a version script to the linker.  This is typically
 used when creating shared libraries to specify additional information
 about the version hierarchy for the library being created.  This option
@@ -6635,7 +6635,7 @@
 @printindex cp

 @tex
-% I think something like @colophon should be in texinfo.  In the
+% I think something like colophon should be in texinfo.  In the
 % meantime:
 \long\def\colophon{\hbox to0pt{}\vfill
 \centerline{The body of this manual is set in}
@@ -6646,7 +6646,7 @@
 \centerline{{\sl\fontname\tensl\/}}
 \centerline{are used for emphasis.}\vfill}
 \page\colophon
-% Blame: doc@cygnus.com, 28mar91.
+% Blame: doc cygnus.com, 28mar91.
 @end tex
 
 
--- binutils/gas/doc/c-arc.texi
+++ binutils/gas/doc/c-arc.texi
@@ -212,7 +212,7 @@
 encodings for use of these instructions according to the specification
 by the user.  The parameters are:
 
-@table @bullet
+table  bullet
 @item @var{name}
 Name of the extension instruction

  
--- binutils/gas/doc/c-mips.texi
+++ binutils/gas/doc/c-mips.texi
@@ -160,10 +160,10 @@
 instructions around accesses to the @samp{HI} and @samp{LO} registers.
 @samp{-no-m4650} turns off this option.
 
-@itemx -m3900
-@itemx -no-m3900
-@itemx -m4100
-@itemx -no-m4100
+itemx -m3900
+itemx -no-m3900
+itemx -m4100
+itemx -no-m4100
 For each option @samp{-m@var{nnnn}}, generate code for the MIPS
 @sc{r@var{nnnn}} chip.  This tells the assembler to accept instructions
 specific to that chip, and to schedule for that chip's hazards.
--- binutils/gas/doc/c-tic54x.texi
+++ binutils/gas/doc/c-tic54x.texi
@@ -108,7 +108,7 @@
 is replaced with x.  At this point, x has already been encountered
 and the substitution stops.
 
-@smallexample @code
+@smallexample code
  .asg   "x",SYM1 
  .asg   "SYM1",SYM2
  .asg   "SYM2",x
@@ -125,14 +125,14 @@
 ambiguous by placing colons on either side of the subsym.  The following
 code: 
 
-@smallexample @code
+@smallexample code
  .eval  "10",x
 LAB:X:  add     #x, a
 @end smallexample
 
 When assembled becomes:
 
-@smallexample @code
+@smallexample code
 LAB10  add     #10, a
 @end smallexample
 
@@ -308,7 +308,7 @@
 of a label or address.  For example, if an address @code{_label} resides
 in extended program memory, the value of @code{_label} may be loaded as
 follows:
-@smallexample @code
+@smallexample code
  ldx     #_label,16,a    ; loads extended bits of _label
  or      #_label,a       ; loads lower 16 bits of _label
  bacc    a               ; full address is in accumulator A
@@ -344,7 +344,7 @@
 performed on @var{string} before assignment.
 
 @cindex @code{eval} directive, TIC54X
-@itemx .eval @var{string}, @var{name}
+itemx .eval var string , var name
 Evaluate the contents of string @var{string} and assign the result as a
 string to the subsym @var{name}.  String replacement is performed on
 @var{string} before assignment.  
--- binutils/gas/doc/as.texinfo
+++ binutils/gas/doc/as.texinfo
@@ -3732,17 +3732,17 @@
 * ABORT::                       @code{.ABORT}
 @end ifset
 
-* Align::                       @code{.align @var{abs-expr} , @var{abs-expr}}
+ Align::                       
 * Altmacro::                    @code{.altmacro}
-* Ascii::                       @code{.ascii "@var{string}"}@dots{}
+ Ascii::                       
 * Asciz::                       @code{.asciz "@var{string}"}@dots{}
 * Balign::                      @code{.balign @var{abs-expr} , @var{abs-expr}}
 * Byte::                        @code{.byte @var{expressions}}
 * Comm::                        @code{.comm @var{symbol} , @var{length} }
 
-* CFI directives::		@code{.cfi_startproc}, @code{.cfi_endproc}, etc.
+ CFI directives::		, etc.
 
-* Data::                        @code{.data @var{subsection}}
+ Data::                        sd
 @ifset COFF
 * Def::                         @code{.def @var{name}}
 @end ifset
@@ -3802,11 +3802,11 @@
 * Line::                        @code{.line @var{line-number}}
 @end ifclear
 
-* Linkonce::			@code{.linkonce [@var{type}]}
+ Linkonce::			
 * List::                        @code{.list}
 * Ln::                          @code{.ln @var{line-number}}
 
-* LNS directives::              @code{.file}, @code{.loc}, etc.
+ LNS directives::              df
 
 * Long::                        @code{.long @var{expressions}}
 @ignore
@@ -3920,8 +3920,8 @@
 @end ifset
 @end ifset
 
-@node Align
-@section @code{.align @var{abs-expr}, @var{abs-expr}, @var{abs-expr}}
+node Align
+section code.align varabs-expr, var abs-expr, varabs-expr
 
 @cindex padding the location counter
 @cindex @code{align} directive
@@ -3964,8 +3964,8 @@
 described later, which have a consistent behavior across all
 architectures (but are specific to GAS).
 
-@node Ascii
-@section @code{.ascii "@var{string}"}@dots{}
+node Ascii
+section code.ascii "varstring"dots
 
 @cindex @code{ascii} directive
 @cindex string literals
@@ -4057,7 +4057,7 @@
 @samp{@var{symbol} .comm, @var{length}}; @var{symbol} is optional.
 @end ifset
 
-@node CFI directives
+node CFI directives
 @section @code{.cfi_startproc}
 @cindex @code{cfi_startproc} directive
 @code{.cfi_startproc} is used at the beginning of each function that
@@ -4113,7 +4113,7 @@
 might use this to add OS-specific CFI opcodes, or generic CFI
 opcodes that GAS does not yet support.
 
-@node LNS directives
+node LNS directives
 @section @code{.file @var{fileno} @var{filename}}
 @cindex @code{file} directive
 When emitting dwarf2 line number information @code{.file} assigns filenames
@@ -4168,7 +4168,7 @@
 The @var{enable} argument should be either 1 or 0, to enable or disable
 this function respectively.
 
-@node Data
+node Data
 @section @code{.data @var{subsection}}
 
 @cindex @code{data} directive
@@ -4820,7 +4820,7 @@
 debugging.
 @end ifclear
 
-@node Linkonce
+node Linkonce
 @section @code{.linkonce [@var{type}]}
 @cindex COMDAT
 @cindex @code{linkonce} directive
